### ISSUE TITLE
does not check if the case file exists

### ISSUE THREAD

**Issue by Mr-2c** [ExtremeFLOW/neko#1599]
I have noticed that neko does not check whether the case file exists and if it does not it throws an error that some keyword was not found in the case file and it could be due to formatting issues in the json file. I think it might be worth checking first whether the file exists to avoid confusions.

**Comment by timfelle** [comment]
Yes, as mentioned by niclas, we should check that the file exists.

Secondly, we actually do not require a case file, but we allow the system to be used programatically. Atleast that is how the setup is done. However, there are a lot of places in the code that depends on the settings, so it is in effect mandatory to have the case file. 

That is a point we should probably have sorted out somehow. Either ensure the code can run without the case files, or remove the option to run without it. @njansson do you have an idea of which would be better here, my intuition is that almost everyone runs with the case files. But as one who are using the system more as a library I would appreciate the move toward more modularity.

**Comment by timofeymukha** [comment]
Yeah, for me the ideal case would even always be having a constructor without JSON, which the JSON constructor eventually calls after parsing the input.

**Comment by njansson** [comment]
> Yeah, for me the ideal case would even always be having a constructor without JSON, which the JSON constructor eventually calls after parsing the input.

This is what we have for `case_t` now, file or json, with file calling json

**Comment by njansson** [comment]
Fixed in #1601
